src/cable/handler.cr

```diff
@@ -45,19 +45,22 @@ module Cable
             connection.receive(message)
           rescue e : KeyError | JSON::ParseException | JSON::SerializableError
             # handle unknown/malformed messages
+            ws_pinger.stop
             socket.close(HTTP::WebSocket::CloseCode::InvalidFramePayloadData, "Invalid message")
             Cable.server.remove_connection(connection_id)
             Cable.settings.on_error.call(e, "Cable::Handler#socket.on_message")
           rescue e : Cable::Connection::UnathorizedConnectionException
             # handle unauthorized connections
             # no need to log them
+            ws_pinger.stop
             socket.close(HTTP::WebSocket::CloseCode::NormalClosure, "Farewell")
             # most of the time, we will have already removed the connection
             # since the connection is rejected before any messages are received
             # but just in case, we will try remove it anyways
             Cable.server.remove_connection(connection_id)
           rescue e : Exception
             # handle all other exceptions
+            ws_pinger.stop
```

Comment:
These may be overkill here, but according to the comment above:

> turns out, if you close socket in this block the socket.on_close blocked is not called 100% of the time so we need to do it manually

I assume if the `on_close` isn't always being called, then we should always ensure this pinger is stopped.